src/transformers/models/switch_transformers/modeling_switch_transformers.py

```diff
@@ -800,6 +800,9 @@ def forward(
         else:
             router_tuple = (torch.tensor([0]),)
 
+        # Send elements in `router_tuple` to the correct device (due to the above if/else statement)
+        router_tuple = tuple(x.to(hidden_states.device) for x in router_tuple)
```

Comment:
Couldn't we just create the `router_tuple` tensor on the device directly instead of creating then moving? 

```suggestion
            router_tuple = (torch.tensor([0], device=hidden_states.device),)
```

Reply:
Yes, you are right. I was thinking so too, but my brain got messed by the if else statement 😭